src/components/content/monitor/Monitor.tsx

```diff
@@ -0,0 +1,282 @@
+                        value: serviceVo.customerServiceName ?? '',
+                        label: serviceVo.customerServiceName ?? '',
+                    };
+                    cusServiceNameList.push(cusServiceName);
+                }
+            });
+            setCustomerServiceNameList(cusServiceNameList);
+        }
+    };
+
+    const handleChangeCustomerServiceName = (selectCustomerServiceName: string) => {
+        form.setFieldsValue({ ServiceId: getServiceId(selectCustomerServiceName) });
+    };
+
+    const getServiceId = (selectCustomerServiceName: string): string => {
+        let currentID: string = '';
+        if (selectCustomerServiceName) {
+            data.forEach((serviceVo) => {
+                if (selectCustomerServiceName === serviceVo.customerServiceName) {
+                    currentID = serviceVo.id;
+                }
+            });
+        }
+        return currentID;
+    };
+
+    const layout = {
+        labelCol: { span: 8 },
+        wrapperCol: { span: 16 },
+    };
+
+    const tailLayout = {
+        wrapperCol: { offset: 8, span: 16 },
+    };
+
+    const onFinish = (values: { ServiceName: string; CustomerServiceName: string; ServiceId: string }) => {
+        queue.clear();
+        if (timer) {
+            clearInterval(timer);
+        }
+        if (!values.ServiceId) {
+            return;
+        }
+        const selectedServiceId = values.ServiceId;
+        setServiceId(selectedServiceId);
+        void MonitorService.getMetrics(selectedServiceId).then((rsp: Metric[]) => {
+            if (rsp.length > 0) {
+                if (queue.isFull()) {
+                    queue.dequeue();
+                }
+                queue.enqueue(rsp);
+                getMonitorType(queue);
+                startFetchMonitorMetricDataTimer(queue, selectedServiceId);
+            }
+        });
+    };
+
+    const startFetchMonitorMetricDataTimer = (queue: Queue<Metric[]>, selectedServiceId: string) => {
+        const newTimer = setInterval(() => fetchMonitorMetricsData(queue, selectedServiceId), 5000);
+        setTimer(newTimer);
+        return () => {
+            if (timer) {
+                clearInterval(timer);
+            }
+        };
+    };
+
+    const fetchMonitorMetricsData = (queue: Queue<Metric[]>, selectedServiceId: string) => {
+        try {
+            void MonitorService.getMetrics(selectedServiceId).then((rsp: Metric[]) => {
+                if (rsp.length > 0) {
+                    const newMetricData: Metric[] = rsp;
+                    if (queue.isFull()) {
+                        queue.dequeue();
+                    }
+                    queue.enqueue(newMetricData);
+                    getMonitorType(queue);
+                }
+            });
+        } catch (error) {
+            console.error(error);
+        }
+    };
+
+    const getMonitorType = (queue: Queue<Metric[]>) => {
```

Comment:
method name can be more explicit. 

Does the method really need a queue? can it not be simply a Metric[] array?

Reply:
fixed.